src/server/index.js

```diff
@@ -0,0 +1,31 @@
+const { createServer } = require( 'http' );
+const healthCheckUrl = '/cache-healthcheck?';
+
+module.exports = ( { PORT = 3000, requestHandler, express = false, logger = console } ) => {
+	if ( requestHandler && express ) {
+		logger.info( 'Creating an Express server...' );
+		requestHandler.get( healthCheckUrl, ( req, res ) => {
+			res.status( 200 ).end();
+		} );
+
+		logger.info( `Starting server on port ${ PORT }...` );
+		return requestHandler.listen( PORT );
```

Comment:
Something else I just thought of - we might want to consider returning an object that has a `.listen()` method, so that the end user can control when listening starts.

For example, for tests, users may want just the server itself, without it binding to any ports (so they can use supertest on it). If we start the listening automatically, then there isn't a way to test without listening.

The alternative would be that you only use the vip go wrapper in production (and not in your tests), but I wonder if that opens us up to too many different code paths between prod and testing/dev, which would could cause inconsistent behavior between test and production.

Reply:
I stumbled upon this when I started writing tests. Having the instance of the app instead of the server itself is the best approach. Even if the client will be responsible for adding `.listen()` but I think it's a good tradeoff.

I didn't go in this direction as I saw that the issue has **Should boot the server**, but if we are ok I can change the behavior to this 👍

Reply:
This is what I'm thinking of. The returned object will be like the following:
``` js
const wrapApplication = ( application, PORT ) => {
	let app = application;
	let server;

	return {
		app,
		server,
		listen: ( connected ) => {
			server = app.listen(PORT, connected)
		},
		close: () => {
			server.close()
		}
	}
}
```
Giving the user access to the app itself and to `.listen()` and `.close()`